examples/src/demos/Physics.js

```diff
@@ -20,20 +20,26 @@ export function Provider({ children }) {
 }
 
 // Custom hook to maintain a world physics body
-export function useCannon({ ...props }, fn, deps = []) {
+export function useCannon({ ...props }, fn) {
   const ref = useRef()
   // Get cannon world object
   const world = useContext(context)
+
   // Instanciate a physics body
-  const [body] = useState(() => new CANNON.Body(props))
+  const [body] = useState(new CANNON.Body(props))
+  const [instanced, setInstanced] = useState(false)
   useEffect(() => {
     // Call function so the user can add shapes
-    fn(body)
+    if (!instanced) {
+      fn(body)
+      setInstanced(true)
+    }
     // Add body to world on mount
     world.addBody(body)
     // Remove body on unmount
     return () => world.removeBody(body)
-  }, deps)
+  }, [instanced, fn, world, body])
+
```

Comment:
The way we we injecting deps fails linting. We actually aren't using that option anyway here, so I removed it to simplify this example. In order to prevent `fn(body)` from being called more than needed, I had to introduce the `instanced` behavior seen here. If you load up the example in the browser, it still works as intended with these changes.

Reply:
i think the demo could even be replaced by one of the use-cannon demos. but that's for later.